pallets/block-rewards/src/lib.rs

```diff
@@ -0,0 +1,475 @@
+	pub inc: BoundedVec<T::AccountId, T::MaxCollators>,
+	pub out: BoundedVec<T::AccountId, T::MaxCollators>,
+}
+
+/// Type that contains the associated data of an session
+#[derive(Encode, Decode, TypeInfo, MaxEncodedLen, PartialEq, Eq, RuntimeDebugNoBound)]
+#[scale_info(skip_type_params(T))]
+pub struct SessionData<T: Config> {
+	/// Amount of rewards per session for a single collator.
+	pub(crate) collator_reward: T::Balance,
+	/// Total amount of rewards per session
+	/// NOTE: Is ensured to be at least collator_reward * collator_count.
+	pub(crate) total_reward: T::Balance,
+	/// Number of current collators.
+	/// NOTE: Updated automatically and thus not adjustable via extrinsic.
+	pub collator_count: u32,
+}
+
+impl<T: Config> Default for SessionData<T> {
+	fn default() -> Self {
+		Self {
+			collator_reward: T::Balance::zero(),
+			total_reward: T::Balance::zero(),
+			collator_count: 0,
+		}
+	}
+}
+
+/// Type that contains the pending update.
+#[derive(
+	PartialEq, Clone, DefaultNoBound, Encode, Decode, TypeInfo, MaxEncodedLen, RuntimeDebugNoBound,
+)]
+#[scale_info(skip_type_params(T))]
+pub struct SessionChanges<T: Config> {
+	pub collators: CollatorChanges<T>,
+	pub collator_count: Option<u32>,
+	collator_reward: Option<T::Balance>,
+	total_reward: Option<T::Balance>,
+}
+
+pub const COLLATOR_GROUP_ID: u32 = 1;
+pub const DEFAULT_COLLATOR_STAKE: u32 = 1000;
+pub const STAKE_CURRENCY_ID: CfgCurrencyId = CfgCurrencyId::Rewards { id: *b"blkrwrds" };
+
+pub(crate) type DomainIdOf<T> = <<T as Config>::Domain as TypedGet>::Type;
+pub(crate) type NegativeImbalanceOf<T> = <<T as Config>::Currency as CurrencyT<
+	<T as frame_system::Config>::AccountId,
+>>::NegativeImbalance;
+
+#[frame_support::pallet]
+pub mod pallet {
+
+	use super::*;
+
+	pub const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		/// Required origin for admin purposes for configuring groups and currencies.
+		type AdminOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+
+		/// Type used to handle balances.
+		type Balance: Balance
+			+ MaxEncodedLen
+			+ FixedPointOperand
+			+ Into<<<Self as Config>::Currency as CurrencyT<Self::AccountId>>::Balance>
+			+ MaybeSerializeDeserialize;
+
+		/// Domain identification used by this pallet
+		type Domain: TypedGet;
+
+		/// Type used to handle group weights.
+		type Weight: Parameter + MaxEncodedLen + EnsureAdd + Unsigned + FixedPointOperand + Default;
+
+		/// The reward system used.
+		type Rewards: GroupRewards<Balance = Self::Balance, GroupId = u32>
+			+ AccountRewards<
+				Self::AccountId,
+				Balance = Self::Balance,
+				CurrencyId = (DomainIdOf<Self>, CfgCurrencyId),
+			> + CurrencyGroupChange<GroupId = u32, CurrencyId = (DomainIdOf<Self>, CfgCurrencyId)>
+			+ DistributedRewards<Balance = Self::Balance, GroupId = u32>;
```

Comment:
I think you do not need this bound if uses the `reward_group()` directly.

Reply:
Thanks! 5f845386ea3cb664fd71aec49f927725afd031ca